File: app/src/main/java/com/legue/axel/lolsummonertool/network/response/item/ItemsResponse.java
package com.legue.axel.lolsummonertool.network.response.item;

import com.google.gson.annotations.SerializedName;
import com.legue.axel.lolsummonertool.database.model.item.ItemGroup;
import com.legue.axel.lolsummonertool.database.model.item.ItemTree;

import java.util.LinkedHashMap;
import java.util.List;

public class ItemsResponse {

    public String type;
    public String version;
    //public Basic basic;
    @SerializedName("data")
    public LinkedHashMap<String, ItemDetailResponse> itemList;
    @SerializedName("groups")
    public List<ItemGroup> groupsList;
    @SerializedName("tree")
    public List<ItemTree> itemTreeList;

    public ItemsResponse() {
    }

    public ItemsResponse(String type, String version, LinkedHashMap<String, ItemDetailResponse> itemList, List<ItemGroup> groupsList, List<ItemTree> itemTreeList) {
        this.type = type;
        this.version = version;
        this.itemList = itemList;
        this.groupsList = groupsList;
        this.itemTreeList = itemTreeList;
    }

}

File: app/src/main/java/com/legue/axel/lolsummonertool/network/response/item/ItemsResponse.kt
package com.legue.axel.lolsummonertool.network.response.item

import com.google.gson.annotations.SerializedName
import com.legue.axel.lolsummonertool.database.model.item.ItemGroup
import com.legue.axel.lolsummonertool.database.model.item.ItemTree
import java.util.*

data class ItemsResponse(
        var type: String,
        var version: String,
        @field: SerializedName("data")
        var itemList: LinkedHashMap<String, ItemDetailResponse>,
        @field: SerializedName("groups")
        var groupsList: List<ItemGroup>,
        @field: SerializedName("tree")
        var itemTreeList: List<ItemTree>
)